pulpcore/tests/functional/api/using_plugin/test_checkpoint.py

```diff
@@ -0,0 +1,193 @@
+"""Tests for checkpoint distribution and publications."""
+
+from datetime import datetime, timedelta
+import re
+from time import sleep
+from urllib.parse import urlparse
+from aiohttp import ClientResponseError
+import pytest
+from pulpcore.client.pulp_file import exceptions
+from pulpcore.content.handler import Handler
+
+
+@pytest.fixture(scope="class")
+def random_content(tmp_path_factory, file_bindings, monitor_task):
+    def _random_content(name):
+        file = tmp_path_factory.mktemp("content") / name
+        file.write_text(name)
+        return monitor_task(
+            file_bindings.ContentFilesApi.create(relative_path=name, file=file).task
```

Comment:
as of pulpcore 3.70 this sadly needs to be
```suggestion
            file_bindings.ContentFilesApi.create(relative_path=name, file=str(file)).task
```